### PR TITLE
Feature/filtering la

### LINKED ISSUE #4756: Implement filter for `La` output to filter over features
When you check the output of `La` (see below) you cannot easily filter over it.

for example it would be nice to show only which arch supports RzIL or has an assembler

```
[0x00000000]> La
_dAeI 8 16       6502        LGPL3   6502/NES/C64/Tamagotchi/T-1000 CPU
adAeI 8          8051        PD      8051 Intel CPU
_dA__ 32         amd29k      LGPL3   AMD 29k RISC CPU (by deroad)
_dA__ 16 32      arc         GPL3    Argonaut RISC Core
adAeI 16 32 64   arm         BSD     Capstone ARM disassembler
[....]
[0x00000000]> La I
_dAeI 8 16       6502        LGPL3   6502/NES/C64/Tamagotchi/T-1000 CPU
adAeI 8          8051        PD      8051 Intel CPU
adAeI 16 32 64   arm         BSD     Capstone ARM disassembler
adAeI 8 16       avr         LGPL3   AVR Atmel
adA_I 16 32 64   bf          LGPL3   Brainfuck (by pancake, nibble) v4.0.0
adAeI 16         gb          LGPL3   GameBoy(TM) (z80-like) (by condret)
_dA_I 32         hexagon     LGPL3   Qualcomm Hexagon (QDSP6) V6 (by Rot127)
_dA_I 16         msp430      LGPL3   msp430 disassembly plugin
_dAeI 16 32      pic         LGPL3   PIC disassembler
_dAeI 32 64      ppc         BSD     Capstone PowerPC disassembler (by pancake)
adAeI 32         sh          LGPL3   SuperH-4 CPU (by DMaroo)
_dA_I 32         tricore     BSD     Siemens TriCore CPU (by billow)
_dAeI 32         v810        LGPL3   v810 disassembly plugin (by pancake)
_dAeI 32         v850        LGPL3   v850 disassembly plugin
_dAeI 16 32 64   x86         BSD     Capstone X86 disassembler
[0x00000000]> La dAeI
_dAeI 8 16       6502        LGPL3   6502/NES/C64/Tamagotchi/T-1000 CPU
adAeI 8          8051        PD      8051 Intel CPU
adAeI 16 32 64   arm         BSD     Capstone ARM disassembler
adAeI 8 16       avr         LGPL3   AVR Atmel
adAeI 16         gb          LGPL3   GameBoy(TM) (z80-like) (by condret)
_dAeI 16 32      pic         LGPL3   PIC disassembler
_dAeI 32 64      ppc         BSD     Capstone PowerPC disassembler (by pancake)
adAeI 32         sh          LGPL3   SuperH-4 CPU (by DMaroo)
_dAeI 32         v810        LGPL3   v810 disassembly plugin (by pancake)
_dAeI 32         v850        LGPL3   v850 disassembly plugin
_dAeI 16 32 64   x86         BSD     Capstone X86 disassembler
[0x00000000]> La dAI
_dAeI 8 16       6502        LGPL3   6502/NES/C64/Tamagotchi/T-1000 CPU
adAeI 8          8051        PD      8051 Intel CPU
adAeI 16 32 64   arm         BSD     Capstone ARM disassembler
adAeI 8 16       avr         LGPL3   AVR Atmel
adA_I 16 32 64   bf          LGPL3   Brainfuck (by pancake, nibble) v4.0.0
adAeI 16         gb          LGPL3   GameBoy(TM) (z80-like) (by condret)
_dA_I 32         hexagon     LGPL3   Qualcomm Hexagon (QDSP6) V6 (by Rot127)
_dA_I 16         msp430      LGPL3   msp430 disassembly plugin
_dAeI 16 32      pic         LGPL3   PIC disassembler
_dAeI 32 64      ppc         BSD     Capstone PowerPC disassembler (by pancake)
adAeI 32         sh          LGPL3   SuperH-4 CPU (by DMaroo)
_dA_I 32         tricore     BSD     Siemens TriCore CPU (by billow)
_dAeI 32         v810        LGPL3   v810 disassembly plugin (by pancake)
_dAeI 32         v850        LGPL3   v850 disassembly plugin
_dAeI 16 32 64   x86         BSD     Capstone X86 disassembler
```

### PR DESCRIPTION
**Pull request by well-mannered-goat**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

I updated the `rz_core_asm_plugins_print` function to handle the input of features argument and also removed the arch argument from `rz_core_asm_plugin_print` while also adding argument to handle the features

- the strings feat and feat2 in `rz_core_asm_plugin_print` were being made and then I concatenate them into feat, my filter is passed as flags argument in the function and if any character of flags is not present in feat the function returns `RZ_CMD_STATUS_OK` rather than printing the output for that `RzAsmPlugin`
- updated the `rz_plugins_asm_print_handler` to handle the `rz_core_plugins_asm_print` accordingly with the number of arguments
- also updated the call of function `rz_core_asm_plugins_print` at other places by passing NULL as flags
- updated `plugins_asm_print_args` to have an optional argument name 'features'
- updated the `cmd_plugins.yaml` to handle an argument for La command

**Test plan**

- Normal working of La
image
- Output for only assemblers
image
- Output for only Analysis with RzIL
image

**Closing issues**

closes #4756

### CODE REVIEW

File: librz/core/casm.c [hunk 1]
```diff
@@ -120,17 +121,24 @@ RZ_API RzCmdStatus rz_core_asm_plugin_print(RzCore *core, RzAsmPlugin *ap, const
...
+		strcpy(feat, "a_");
 	}
 	if (!ap->assemble && ap->disassemble) {
-		feat = "_d";
+		strcpy(feat, "_d");
 	}
 	feat2 = has_esil(core, ap->name);
+	strcat(feat, feat2);
+
+	for (int i = 0;flags && flags[i] != '\0'; i++) {
```
Comment (wargio) [inline librz/core/casm.c]: I like how you solved it, but this is not clang formatted. Run sys/clang-format.py (requires clang format) and submit the changes.

**Review by wargio — CHANGES_REQUESTED** [review]
Add a test under test/db/cmd/

**Comment by well-mannered-goat** [comment]
So I have been working on the testing of this command and found out that `Laj` command expected wrong output with only `a` and `d` features in output as initially only those were printed in JSON output.
I updated it to have others too.

Also since I have applied filtering on Laq and Laj commands too, should I do extensive testing for each filtering and their combinations with La command output types?

**Comment by wargio** [comment]
Just add a test for each j and q variant.

**Review by wargio — CHANGES_REQUESTED** [review]
Looks like you did not use the correct clang-format version. Google clang format static and use clang format 16

**Comment by wargio** [comment]
Also for the tests, just merge them into one test.
So it would run multiple commands in one rizin session

**Comment by well-mannered-goat** [comment]
Sorry, would apply the changes as soon as possible

**Comment by wargio** [comment]
take your time. no rush

**Comment by wargio** [comment]
Ps: check the options of sys/clang-format.py to pass the correct version of clang format

**Comment by well-mannered-goat** [comment]
I cant pass the version in the script it seems

I have installed 16.0.0 on my machine

image

image

is this the correct version needed?

**Comment by wargio** [comment]
No, i meant the binary location of clang-format16
-C should allow this

**Comment by well-mannered-goat** [comment]
I passed the path to binary location of clang-format in -c argument and there was no change in any file
Also i did not push the output_file in test directory, do I need to push it too?

**Comment by wargio** [comment]
this is how i run it `./sys/clang-format.py -C /opt/clang-format-static/clang-format-16`, also is upper case `C`

**Comment by well-mannered-goat** [comment]
So this is my binary location
image

this is what i run `./sys/clang-format.py -C /opt/clang-16.0.0/bin/clang-format`

**Review by wargio — CHANGES_REQUESTED** [review]
Ok, just manually revert the changes of
- `librz/arch/isa/rx/rx_opcode_detail.c`
- `librz/bin/format/elf/glibc_elf.h`
- `librz/hash/algorithms/ssdeep/fnv_hash.h`
- `librz/include/rz_magic.h`

Then squash and we can merge.

**Comment by well-mannered-goat** [comment]
squashed the commits
sorry didnt know had to keep less number of commits, would keep in mind about it.

**Comment by wargio** [comment]
> squashed the commits sorry didnt know had to keep less number of commits, would keep in mind about it.

Not really, we normally always squash, but is just to have a clean history here (for readability)

**Comment by well-mannered-goat** [comment]
Seems like RzIL support for xtensa is added

would update the tests

**Review by wargio — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Pull request merged**